.tekton/devfile-sample-go-basic-pull-request.yaml

```diff
@@ -236,6 +237,69 @@ spec:
       workspaces:
       - name: source
         workspace: workspace
+    - name: build-container-arm64
+      params:
+        - name: IMAGE
+          value: $(params.output-image)-arm64
+        - name: DOCKERFILE
+          value: $(params.dockerfile)
+        - name: CONTEXT
+          value: $(params.path-context)
+        - name: HERMETIC
+          value: $(params.hermetic)
+        - name: PREFETCH_INPUT
+          value: $(params.prefetch-input)
+        - name: IMAGE_EXPIRES_AFTER
+          value: $(params.image-expires-after)
+        - name: COMMIT_SHA
+          value: $(tasks.clone-repository.results.commit)
+        - name: ARCH
+          value: arm64
+      runAfter:
+        - build-container-amd64
+      taskRef:
+        resolver: git
+        params:
+          - name: url
+            value: https://github.com/stuartwdouglas/build-definitions.git
+          - name: revision
+            value: multiarch
+          - name: pathInRepo
+            value: task/buildah-remote/0.1/buildah-remote.yaml
+      when:
+        - input: $(tasks.init.results.build)
+          operator: in
+          values:
+            - "true"
+      workspaces:
+        - name: source
+          workspace: workspace
+
+    - name: build-container
+      params:
+        - name: IMAGE
+          value: $(params.output-image)
```

Comment:
This must be using a tagging convention to find the arch-specific images, right? Would it make sense to instead provide a list based on the results from the previous build tasks? e.g.

```yaml
params:
  - name: IMAGE_MANIFESTS
    value:
      - $(tasks.build-container-amd64.results.IMAGE_URL)@$(tasks.build-container-amd64.results.IMAGE_DIGEST)
      - $(tasks.build-container-arm64.results.IMAGE_URL)@$(tasks.build-container-arm64.results.IMAGE_DIGEST)
```

The above uses an [array-type](https://tekton.dev/docs/pipelines/tasks/#array-type) parameter.

Some of the benefits of this approach are:
1. Exact digest is used. This is more secure and deterministic.
2. Task does not rely on users having to follow a certain convention.

Reply:
The downside of this approach is that the task must inspect the config of each image to determine its architecture and os.

Reply:
That could work, I can definitely see the argument about being more secure, especially if the user kicks off two pipelines at once.